packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -27,18 +31,58 @@ export const Appender = forwardRef( ( props, ref ) => {
 		};
 	}, [] );
 
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const setAttributes =
+		( insertedBlockClientId ) => ( _updatedAttributes ) => {
+			updateBlockAttributes( insertedBlockClientId, _updatedAttributes );
+		};
+
+	let maybeLinkUI;
+
+	if ( insertedBlock ) {
+		const link = {
+			url: insertedBlock.attributes.url,
+			opensInNewTab: insertedBlock.attributes.target === '_blank',
+		};
+		maybeLinkUI = (
+			<LinkUI
+				clientId={ insertedBlock.clientId }
+				value={ link }
+				linkAttributes={ {
+					type: insertedBlock.attributes.type,
+					url: insertedBlock.attributes.url,
+					kind: insertedBlock.attributes.kind,
+				} }
```

Comment:
We should probably make this one prop here and then split it out into `value` to pass down into `<LinkControl>`.